src/main/java/org/commcare/formplayer/application/FormController.java

```diff
@@ -95,7 +118,25 @@ public FormEntryResponseBean changeLocale(@RequestBody ChangeLocaleRequestBean c
     @UserRestore
     @ConfigureStorageFromSession
     public FormEntryResponseBean answerQuestion(@RequestBody AnswerQuestionRequestBean answerQuestionBean,
-                                                @CookieValue(name = Constants.POSTGRES_DJANGO_SESSION_ID, required = false) String authToken) throws Exception {
+            @CookieValue(name = Constants.POSTGRES_DJANGO_SESSION_ID, required = false) String authToken)
+            throws Exception {
+        return saveAnswer(answerQuestionBean, null);
+    }
+
+    @RequestMapping(value = Constants.URL_ANSWER_MEDIA_QUESTION, method = RequestMethod.POST)
+    @UserLock
+    @UserRestore
+    @ConfigureStorageFromSession
+    public FormEntryResponseBean answerMediaQuestion(
+            @RequestParam("file") MultipartFile file,
+            @RequestBody AnswerQuestionRequestBean answerQuestionBean,
```

Comment:
I think this should be using `@RequestParam("answer")` since the whole body will be the multipart request

Reply:
Yes @orangejenny  did flag this to me and I am definitely confused between the request format here.  Looks like you are saying there will be 2 parts in this request - The media file and the answer Json which makes sense to me. 

Although I am testing this with JS side of code [here](https://github.com/dimagi/commcare-hq/compare/jls/web-apps-upload#diff-db36924530834745eae7459c406c8e8bdae13c959bc24f2ab93fe9db18a033f0R126) although it seems like JS is currently sending a part for every property of `AnswerQuestionRequestBean`. This is how the payload looks like in chrome console - 

````
answer
------WebKitFormBoundaryUpptPtsoIV8XP82O
Content-Disposition: form-data; name="ix"

0
------WebKitFormBoundaryUpptPtsoIV8XP82O
Content-Disposition: form-data; name="answer"

sas
------WebKitFormBoundaryUpptPtsoIV8XP82O
Content-Disposition: form-data; name="answersToValidate"

[object Object]
------WebKitFormBoundaryUpptPtsoIV8XP82O
Content-Disposition: form-data; name="oneQuestionPerScreen"

false
------WebKitFormBoundaryUpptPtsoIV8XP82O
Content-Disposition: form-data; name="form_context"

undefined
------WebKitFormBoundaryUpptPtsoIV8XP82O
Content-Disposition: form-data; name="domain"

shubham
------WebKitFormBoundaryUpptPtsoIV8XP82O
Content-Disposition: form-data; name="username"

....
````

Is this wrong ? If yes, is it feasible to change it on HQ side so that there are only ever 2 parts as mentioned above.

One other question, it seems like on the JS side we are going to modify all answer requests to be multipart requests instead of just the media answer ones. As such it doesn't seem to make sense to have 2 separate endpoints one for media and one for normal answers ? We will need to support the earlier format to start with as well though I think we can just have one endpoint `answer` and remove `answer_media` ?

Reply:
> Is this wrong ? If yes, is it feasible to change it on HQ side so that there are only ever 2 parts as mentioned above.

Just saw [this](https://github.com/dimagi/commcare-hq/pull/32026#discussion_r947817490) which resolves this question for me.